public/app/features/datasources/components/EditDataSource.tsx

```diff
@@ -210,7 +211,7 @@ export function EditDataSourceView({
           <div key={extension.id}>
             <Component
               context={{
-                dataSource: dataSource,
+                dataSource: cloneDeep(dataSource),
```

Comment:
nit: not a big fan of comments but in this case maybe we should add a small note